admin/admin-model.php

```diff
@@ -58,11 +58,11 @@ public function add_language( $args ) {
 			update_option( 'polylang', $this->options );
 		}
 
-		$this->clean_languages_cache(); // Update the languages list now !
+		// Refresh languages.
+		$this->clean_languages_cache();
+		$this->get_languages_list();
 
-		// Init a mo_id for this language
-		$mo = new PLL_MO();
-		$mo->export_to_db( $this->get_language( $args['slug'] ) );
+		flush_rewrite_rules(); // Refresh rewrite rules.
```

Comment:
OK to move the action at the end, but for consistency, you should move `pll_update_language` too.